### ISSUE TITLE
API for working with XML data isn't very intuitive

### ISSUE THREAD

**Issue by danmichaelo** [edsu/pymarc#73]
Having some xml data,
```python
data = open('test.xml', 'rb')
```
, I expected from the README example to be able to do something like

```python
from pymarc import MARCReader
for record in MARCReader(data):
    ...
```

but instead I had to do

```python
import pymarc
for record in pymarc.parse_xml_to_array(data):
    ...
```

Determining the file type should be quite easy from reading the first characters of the file stream: xml if "<?xml", json if "{", plain marc otherwise.

Next I wanted to try to serialize a record to XML. The `Record` object has methods like `as_marc()`, `as_marc21()` and hm, even `as_json()`, but no `as_xml()`! Instead:

```python
pymarc.record_to_xml(record)
```

**Comment by edsu** [comment]
Yeah, that's a fair criticism. Still, you figured it out -- so maybe it's not so bad?

**Comment by danmichaelo** [comment]
We all do ;) But then there's library systems… 

Anyways, I won't be offended if you close this as "wontfix", but consider adding a short xml example to the README first. Might help save others some time.

**Comment by edsu** [comment]
I'll leave this open until one of those things happen. Thanks @danmichaelo !

**Comment by rlskoeser** [comment]
Seconding that reading xml is not intuitive.  Had to look at the test code and this issue before I got it to work.

**Comment by edsu** [comment]
Just out of curiosity would people be ok with:

```python
for record in MARCReader(open('batch.xml', 'rb')):
    # do something useful with a Record
```

creating an in memory array for all the records, and then allowing the iteration to start? I think it would be preferable if it did function as an iterator, but I'm not quite sure how that would combine with the SAX parsing that's going on.
